packages/sdk/src/builder.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { isValidBuilder } from "./builder";
+
+describe("builder", () => {
+  describe("isValidBuilder", () => {
+    // Valid cases
+
+    it("returns true for simple examples", () => {
+      expect(isValidBuilder("myorg/super-optimizer:0.1.2")).toEqual(true);
+      expect(isValidBuilder("myorg/super-optimizer:42")).toEqual(true);
+    });
+
+    it("supports images with no organization", () => {
+      // from https://hub.docker.com/_/ubuntu
+      expect(isValidBuilder("ubuntu:xenial-20200212")).toEqual(true);
+      // from https://hub.docker.com/_/rust
+      expect(isValidBuilder("rust:1.40.0")).toEqual(true);
+    });
+
+    it("supports images with multi level names", () => {
+      expect(isValidBuilder("myorg/department-x/office-y/technology-z/super-optimizer:0.1.2")).toEqual(true);
+    });
+
+    it("returns true for tags with lower and upper chars", () => {
+      expect(isValidBuilder("myorg/super-optimizer:0.1.2-alpha")).toEqual(true);
+      expect(isValidBuilder("myorg/super-optimizer:0.1.2-Alpha")).toEqual(true);
+    });
+
+    it("allows very long images", () => {
+      // This is > 2 KiB of data
+      expect(
+        isValidBuilder(
```

Comment:
should we prevent this?

Reply:
Not as part of the regexp, but I'd set a max length limit on the string.